crates/proto/src/dnssec/dnssec_dns_handle/nsec3_validation.rs

```diff
@@ -838,3 +831,97 @@ fn wildcard_based_encloser_proof<'a>(
         closest_encloser_wildcard: wildcard_encloser,
     }
 }
+
+/// A base32-encoded hash, from the leftmost label of an NSEC3 record owner name.
+struct Nsec3Base32Hash(Vec<u8>);
```

Comment:
Do you think this would be simpler/less duplication if this type wrapped a `Cow<'a, [u8]>`? It would be nice if `compare_ignore_case()` could be inlined into one specific `Ord` impl.